### PR TITLE
Some optimisations to boostHistHelpers and to the helicity scale uncertainty hist

### PR DESCRIPTION
**Pull request by kdlong**
* Avoid some copies in boostHistHelpers when using createNew=False
* Reorganise the helicity hist to variation function to minimise copies
For the nominal analysis the speed up is about 10% (mostly coming from the helicity function), it might be better for the unfolding

### CODE REVIEW

**Comment by kdlong** [comment]
It looks like I've messed up the divide function, probably in the cutoff parameter

File: utilities/boostHistHelpers.py [hunk 1]
```diff
@@ -6,72 +6,56 @@
...
+    h1vals,h2vals,h1vars,h2vars = valsAndVariances(h1, h2)
     # By the argument that 0/0 = 1
     out[(np.abs(h2vals) < cutoff) & (np.abs(h1vals) < cutoff)] = 1.
     val = np.divide(h1vals, h2vals, out=out, where=np.abs(h2vals)>cutoff)
 
-    if h1._storage_type() != hist.storage.Weight() or h2._storage_type() != hist.storage.Weight():
-        newh = hist.Hist(*outh.axes, data=val)
-        return newh
-    else:
+    outh.values(flow=True)[...] = val
```
Comment (davidwalter2) [inline utilities/boostHistHelpers.py:38]: This line sets the content of h1vals and h2vals to val (if createNew=False), which is not what we want, so ```outh.values(flow=True)[...] = val``` has to go just before the return

**Review by davidwalter2 — COMMENTED** [review]
I think I found the problem with the uncertainties in the divideHists function

File: utilities/boostHistHelpers.py [hunk 2]
```diff
@@ -6,72 +6,56 @@
...
+    # Transpose because we keep syst axis last, but numpy broadcasts from the front
+    new_vals = np.broadcast_to(h1.view(flow=True).T, h2.view(flow=True).T.shape).T
+
+    return hist.Hist(*h2.axes, data=new_vals, storage=h1._storage_type())
 
 # returns h1/h2
-def divideHists(h1, h2, cutoff=1e-5, allowBroadcast=True, rel_unc=False, createNew=True):
-    # To get the broadcast shape right
-    outh = h1 if not allowBroadcast else broadcastOutHist(h1, h2)
+def divideHists(h1, h2, cutoff=1e-5, allowBroadcast=True, rel_unc=False, createNew=False):
```
Comment (davidwalter2) [inline utilities/boostHistHelpers.py:23]: Should we maybe keep createNew=True because setting it to False could destroy backward compatibility.

Comment (kdlong) [reply utilities/boostHistHelpers.py]: Yes probably

**Comment by davidwalter2** [comment]
The PR validation has some significant differences between the impact plots for example, and also for the uncertainty on the fakes, hmm ...

**Comment by kdlong** [comment]
Yeah, I mixed in the combined fit updates here. I'm not quite sure if the differences are expected for that, but let's focus on that PR first (#229)

**Comment by kdlong** [comment]
Closing in favour of #238, which should be easier to review

**Pull request closed without merging**